libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/relations/many_to_many/implicit.rs

```diff
@@ -0,0 +1,79 @@
+use datamodel_connector::{ConnectorCapability, DatamodelError};
+use itertools::Itertools;
+use parser_database::walkers::ImplicitManyToManyRelationWalker;
+
+use crate::transform::ast_to_dml::validation_pipeline::context::Context;
+
+/// Our weird many-to-many requirement.
+pub(crate) fn validate_singular_id(relation: ImplicitManyToManyRelationWalker<'_>, ctx: &mut Context<'_>) {
+    for relation_field in [relation.field_a(), relation.field_b()].iter() {
+        if !relation_field.related_model().has_single_id_field() {
+            let message = format!(
+                "The relation field `{}` on Model `{}` references `{}` which does not have an `@id` field. Models without `@id` cannot be part of a many to many relation. Use an explicit intermediate Model to represent this relationship.",
+                &relation_field.name(),
+                &relation_field.model().name(),
+                &relation_field.related_model().name(),
+            );
+
+            ctx.push_error(DatamodelError::new_field_validation_error(
+                &message,
+                relation_field.model().name(),
+                relation_field.name(),
+                relation_field.ast_field().span,
+            ));
+
+            continue;
+        }
+
+        if !relation_field.references_singular_id_field() {
+            ctx.push_error(DatamodelError::new_validation_error(
+            format!(
+                "Many to many relations must always reference the id field of the related model. Change the argument `references` to use the id field of the related model `{}`. But it is referencing the following fields that are not the id: {}",
```

Comment:
```suggestion
                "Implicit many to many relations must always reference the id field of the related model. Change the argument `references` to use the id field of the related model `{}`. But it is referencing the following fields that are not the id: {}",
```

Reply:
tbd but I think it's all many-to-many relations

Reply:
...that was agreed on

Reply:
Don't know who to believe then :D https://github.com/prisma/prisma-engines/pull/2671#discussion_r804113981

Reply:
I think you might actually be right now. If this also applies to the new MongoDB relation, that one is not really _implicit_ as it has the explicit arrays for storing the references.